src/ng/rootScope.js

```diff
@@ -900,6 +909,13 @@ function $RootScopeProvider() {
         this.$on = this.$watch = this.$watchGroup = function() { return noop; };
         this.$$listeners = {};
 
+        var that = this;
+        if (msie === 9) {
+          this.$evalAsync(function() {
+            cleanUpScope(that);
```

Comment:
So we can just set `that.$$nextSibling = null` before calling `cleanUpScope(that)` to prevent `cleanUpScope` from trying to kill sibling nodes.